Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -259,13 +259,32 @@ extension BoringSSLRSAPublicKey {
                     switch padding.backing {
                     case .pkcs1_oaep: rawPadding = RSA_PKCS1_OAEP_PADDING
                     }
-                    let rc = CCryptoBoringSSLShims_RSA_public_encrypt(
-                        CInt(dataPtr.count),
-                        dataPtr.baseAddress,
+
+                    let pkey = CCryptoBoringSSL_EVP_PKEY_new()
+                    defer {
+                        CCryptoBoringSSL_EVP_PKEY_free(pkey)
+                    }
+
+                    CCryptoBoringSSL_EVP_PKEY_set1_RSA(pkey, self.pointer)
+
+                    // nil engine defaults to the standard implementation with no hooks
+                    let ctx = CCryptoBoringSSL_EVP_PKEY_CTX_new(pkey, nil)
+                    defer {
+                        CCryptoBoringSSL_EVP_PKEY_CTX_free(ctx)
+                    }
+
+                    CCryptoBoringSSL_EVP_PKEY_encrypt_init(ctx)
+                    CCryptoBoringSSL_EVP_PKEY_CTX_set_rsa_padding(ctx, rawPadding)
+
+                    var writtenLength = bufferPtr.count
```

Comment:
We don't currently do anything with this - we could use it to check if the length meets our expectations and precondition/throw. WDYT @Lukasa ?

Reply:
https://commondatastorage.googleapis.com/chromium-boringssl-docs/evp.h.html#EVP_PKEY_encrypt_init
> EVP_PKEY_encrypt encrypts in_len bytes from in. If out is NULL, the maximum size of the ciphertext is written to out_len. Otherwise, *out_len must contain the number of bytes of space available at out. If sufficient, the ciphertext will be written to out and *out_len updated with the true length.

My interpretation of the above was that `out_len`/`writtenLength` should contain the buffer length and not 0.

Reply:
Nice catch.